subprojects/instant-execution/src/integTest/groovy/org/gradle/instantexecution/InstantExecutionIntegrationTest.groovy

```diff
@@ -104,6 +104,28 @@ class InstantExecutionIntegrationTest extends AbstractIntegrationSpec {
         result.assertTasksExecuted(":a")
     }
 
+    def "can invalidate instant execution cache from the command line with --refresh-dependencies"() {
+        given:
+        buildFile << """
+            println("classic")
+        """
+
+        when:
+        instantRun "help"
+
+        and:
+        instantRun "help"
+
+        then:
+        outputDoesNotContain("classic")
+
+        when:
+        instantRun "help", "--refresh-dependencies"
+
+        then:
+        outputContains("classic")
```

Comment:
⭕️ I think I would prefer a test that would actually change the cached state and assert that not only `--refresh-dependencies` causes build logic to run but that the cached state is then different after it

Reply:
It's too early in the spike to implement that as part of this pr, let's do it soon enough

Reply:
Proper test added, using a custom task type worked until custom task actions are supported